src/solvers/smt/smt_overflow.cpp

```diff
@@ -200,25 +200,42 @@ smt_astt smt_convt::overflow_cast(const expr2tc &expr)
 
 smt_astt smt_convt::overflow_neg(const expr2tc &expr)
 {
-  // If in integer mode, this is completely pointless. Return false.
+  // If in integer mode, this check is irrelevant, return false
   if (int_encoding)
     return mk_smt_bool(false);
 
   // Extract operand
   const overflow_neg2t &neg = to_overflow_neg2t(expr);
   unsigned int width = neg.operand->type->get_width();
 
-  // Ensure the operand is a signed integer type
-  expr2tc operand = typecast2tc(signedbv_type2tc(width), neg.operand);
+  // Check if operand is unsigned
+  bool is_unsigned = is_unsignedbv_type(neg.operand->type);
 
-  // Simplify if typecast was performed
-  simplify(operand);
+  if (is_unsigned)
+  {
+    // **Unsigned Negation Overflow Check**
+    // In unsigned arithmetic, negation (-x) is effectively (UINT_MAX + 1 - x).
+    // Any nonzero value negated will wrap around, which is unexpected behavior.
+    expr2tc zero = constant_int2tc(neg.operand->type, 0);
+    // Overflow occurs if operand is x == 0
```

Comment:
It's the other way around, isn't it?
Overflow occurs <=> x != 0
E.g. UBSAN agrees with me:
https://godbolt.org/z/YKhvenec7